### ISSUE TITLE
txid não exibido em alguns bancos

### ISSUE THREAD

**Issue by cunhamc** [bacen/pix-api#511]
Pessoal,
a informação que é enviada no campo txid nos PIX Estáticos ele não está sendo visualizado nos extratos de alguns bancos. Alguem já teve problema desse tipo?
No caso que eu pude verificar foram nos bancos Itau e Inter. Ambos as contas são PJ e a única informação que o recebedor visualiza é o mensagem do pagador que não é incluida no QRCode do PIX estático que foi gerado pelo recebedor.

**Comment by rubenskuhl** [comment]
No C6, tanto no extrato quanto no comprovante específico da transação, não é citado o txid... apenas o e2eid. E imagino que isso seja comum em vários bancos e não limitado a QR-Codes estáticos. Por isso em nosso sistema de atendimento possibilitamos ao nosso agente pesquisar tanto pelo txid quanto pelo e2eid para localizar a transação.

**Comment by cunhamc** [comment]
Desculpe a ignorancia, mas qual seria a informação do campo e2eid? seria a Identificação da Transação? Se for eu entendo que não seria possivel identificar o pagamento de uma cobrança específica pq essa informação é gerada somente no momento do pagamento e não quando estou gerando a cobrança. Isso usando o é qr-code estático.

Existe alguma solução pra isso no qual eu seja possivel apenas usando o extrato identificar o pagamento de uma cobrança gerada por qrcode estático?

**Comment by rubenskuhl** [comment]
Sim, o End-2-End-Id é identificado usualmente como Identificação da Transação. E não tem problema só ser gerado no pagamento, pois você recebe o e2eid no webhook e no GET de /pix. Basta que o QR-Code estático tenha txid para ele ser sinalizado no webhook, e aí você receberá tanto o txid quanto o e2eid no webhook e no GET de /pix. O GET de /pix pode ser filtrado tanto por e2eid quanto por inicio/fim/txid, dependendo de qual deles você tiver para buscar a informação.

**Comment by cunhamc** [comment]
no Santander e PicPay (que foram um dos que fiz teste) é exibido essa informação no extrato com o nome "Identificação do Pagamento", porem no Itau e Inter não.

O Cenário é o seguinte:
A empresa aqui recebe doações em dinheiro e, até então, era somente compartilhado a chave pix para as pessoas fazerem suas doações.
O problema é que essa empresa recebe doações com destinos específicos (exemplo, reforma das salas, compra de materiais, cestas básicas etc.) e para saber isso eles pedem para as pessoas colocar esse comentário quando estão fazendo o pix, mas essa informação nem sempre vem ou quando vem estão sem padrão no texto e dificulta a identificação.
A minha ideia foi no site deles dar a opção de escolher para onde vai sua doação e ao fazer isso é gerado um QRCode com essa informação no txid (que foi o único campo que é possivel já deixar preenchido na geração do QRCode), assim evita o problema que coloquei acima, mas se a informação não chegar para eles conferirem no extrato não vai adiantar em nada.

A ideia é simples, o transtorno não é critico e acaba não justificando contratar um PSP apenas para isso. Mas foi uma implementação bem simples e por isso achei que valeria a pena.

**Comment by arnaldojnr** [comment]
> Bom dia, alguém sabe se no extrato do banco do brasil é informado o txid?

Informa sim, acabei de fazer o teste
